content/docs/command-reference/checkout.md

```diff
@@ -10,21 +10,26 @@ usage: dvc checkout [-h] [-q | -v] [--summary] [-d] [-R] [-f]
                     [--relink] [targets [targets ...]]
 
 positional arguments:
-  targets        Limit command scope to these stages or .dvc files.
-                 Using -R, directories to search for stages or .dvc
-                 files can also be given.
+  targets       Limit command scope to these tracked files/directories,
+                .dvc files, or stage names.
 ```
 
 ## Description
 
-`.dvc` and `dvc.lock` [files](/doc/user-guide/dvc-files-and-directories) act as
-pointers to specific version of data files or directories tracked by DVC. This
-command synchronizes the workspace data with the versions specified in the
-current `.dvc` and `dvc.lock` files.
+Synchronizes the workspace data with all the
+[tracked](/doc/command-reference/add) files and directories specified in the
+present `dvc.lock` and `.dvc` files, which act as pointers to corresponding
+<abbr>cached</abbr> data contents.
```

Comment:
Qq @efiop: Is it correct to talk about dvc.lock instead of dvc.yaml in this reference? And probably others such as fetch. I.e. which one gets scanned? I'm assuming even if dvc.yaml is scanned, the stage needs to exist in dvc.lock, otherwise it's discarded from the checkout/fetch/etc targets. Thanks

Reply:
P/s I guess it's out of scope but it would be good to know anyway, so I can address here or in another PR.

Reply:
No, it is incorrect. `dvc.yaml` is the source of all truth, so we look there first and then `dvc.lock` second. If something is not in the `dvc.yaml` -- it won't be checked out, even if it is present in dvc.lock.

Reply:
Got it. I had a feeling that may be the way it's implemented. But I'm still wondering: in effect, does it have to be in dvc.lock? Let me rephrase: if it's in dvc.yaml, but not found in dvc.lock, does it get skipped anyway for certain purpose? e.g. checkout, fetch, etc. which need to find the exact file name in cache or remote storage. Thanks

Reply:
If something is not in dvc.lock it will be skipped (with a warning), because there is nothing we can do without the hash.

Reply:
OK, updated a bit in ff3d7aa5. Resolving, thanks!

Reply:
So, they are both important. I'm not sure what is the best way to describe it yet, but it feels that in some cases we need to be clear that the list of files/directories to synchronize is taken from `.dvc` + `dvc.yaml` and actual versions are defined in `.dvc` + `dvc.lock`. Depending on a context and a place we should mention one or both to my mind.